dCommon/GeneralUtils.h

```diff
@@ -134,6 +134,11 @@ namespace GeneralUtils {
 		return T();
 	}
 
+	template <typename T>
+	inline T GenerateRandomNumber() {
+		return GenerateRandomNumber<T>(std::numeric_limits<T>::min(), std::numeric_limits<T>::max());
```

Comment:
Could have a static_assert here as the above does to ensure we are getting a valid integer type